src/apple/disk.rs

```diff
@@ -53,14 +62,330 @@ impl DiskExt for Disk {
+    unsafe {
+        get_int_value(
+            disk_props.inner(),
+            DictKey::Extern(ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey),
+        )
+        .filter(|bytes| *bytes != 0)
+        .or_else(|| {
+            get_int_value(
+                disk_props.inner(),
+                DictKey::Extern(ffi::kCFURLVolumeAvailableCapacityKey),
+            )
+        })
+    }
+    .unwrap() as u64
+}
+
+pub(super) enum DictKey {
+    Extern(CFStringRef),
+    #[cfg(target_os = "macos")]
+    Defined(&'static str),
+}
+
+unsafe fn get_dict_value<T, F: FnOnce(*const c_void) -> Option<T>>(
+    dict: CFDictionaryRef,
+    key: DictKey,
+    callback: F,
+) -> Option<T> {
+    #[cfg(target_os = "macos")]
+    let _defined;
+    let key = match key {
+        DictKey::Extern(val) => val,
+        #[cfg(target_os = "macos")]
+        DictKey::Defined(val) => {
+            _defined = CFReleaser::new(cfs::CFStringCreateWithBytesNoCopy(
+                kCFAllocatorDefault,
+                val.as_ptr(),
+                val.len() as isize,
+                cfs::kCFStringEncodingUTF8,
+                false as _,
+                core_foundation_sys::base::kCFAllocatorNull,
+            ))
+            .unwrap();
+
+            _defined.inner()
+        }
+    };
+
+    let mut value = std::ptr::null();
+    if CFDictionaryGetValueIfPresent(dict, key.cast(), &mut value) != 0 {
+        callback(value)
+    } else {
+        None
+    }
+}
+
+pub(super) unsafe fn get_str_value(dict: CFDictionaryRef, key: DictKey) -> Option<String> {
+    get_dict_value(dict, key, |v| {
+        let v = v as cfs::CFStringRef;
+
+        let len_utf16 = cfs::CFStringGetLength(v);
+        let len_bytes = len_utf16 as usize * 2; // Two bytes per UTF-16 codepoint.
+
+        let v_ptr = cfs::CFStringGetCStringPtr(v, cfs::kCFStringEncodingUTF8);
+        if v_ptr.is_null() {
+            // Fallback on CFStringGetString to read the underlying bytes from the CFString.
+            let mut buf = vec![0; len_bytes];
+            let success = cfs::CFStringGetCString(
+                v,
+                buf.as_mut_ptr(),
+                len_bytes as _,
+                cfs::kCFStringEncodingUTF8,
+            );
+
+            if success != 0 {
+                utils::vec_to_rust(buf)
             } else {
-                false
+                None
             }
+        } else {
+            utils::cstr_to_rust_with_size(v_ptr, Some(len_bytes))
+        }
+    })
+}
+
+unsafe fn get_bool_value(dict: CFDictionaryRef, key: DictKey) -> Option<bool> {
+    get_dict_value(dict, key, |v| Some(v as CFBooleanRef == kCFBooleanTrue))
+}
+
+unsafe fn get_int_value(dict: CFDictionaryRef, key: DictKey) -> Option<i64> {
+    get_dict_value(dict, key, |v| {
+        let mut val: i64 = 0;
+        if CFNumberGetValue(
+            v.cast(),
+            core_foundation_sys::number::kCFNumberSInt64Type,
+            &mut val as *mut i64 as *mut c_void,
```

Comment:
It comes from two things:
 * The function header might change (unlikely though but still something I'm wary of).
 * Coherency with the rest.

A very small point in favour of generic casting too: it's a bit shorter (no need to import `c_void` too).

But in any case, if you don't want to change it, I won't be too annoying for this one as it is very unlikely it'll ever change. Mostly a personal preference at this point.

Reply:
I think that this one is worth keeping as-is for the following reasons:
- We already need to import `c_void` for other functions in this file, so we don't save an import
- CoreFoundation objects are always typed as `c_void` throughout the headers and it would be a huge and unlikely change to move away from that. 
- This function also needs to take an arbitrary "number-like" CF value so it makes sense that it needs to be type erased regardless
- `val` will always be an `i64` since we declare the value here in Rust.

Reply:
Ok, makes sense. :+1: